### ISSUE TITLE
Threadize face recognition in case of multiple faces

### ISSUE THREAD

**Issue by alessiosavi** [alessiosavi/PyRecognizer#20]
Let every prediction run in parallel in case of multi faces photo

**Comment by ClashLuke** [comment]
Python threading/multiprocessing/joblib all use only one core (but can be loadbalanced to different cores by the OS), how'd you want to implement this? Taskset the affinity?

**Comment by alessiosavi** [comment]
Hi @ClashLuke, thank you for the interest!

Unfortunately, I've just realized (as expected), that is no necessary to parallelize the `predict` phase, cause the most part of the work have to be done during the face encodings. https://github.com/alessiosavi/PyRecognizer/blob/1f3b14be6c795096b92933ddc4c4363dd3ec61c3/datastructure/Classifier.py#L325)

I've used a photos with 103 faces, and the prediction (after that all the faces was encoded) take few milliseconds. Instead, the encodings phase lasts approx 30 sec.

I'm going to close the issue cause is something related to:
- Hardware (GPU)
- Backend library used (dlib)
